mutable_tree.go

```diff
@@ -1028,53 +1028,57 @@ func (tree *MutableTree) balance(node *Node) (newSelf *Node, err error) {
 // saveNewNodes save new created nodes by the changes of the working tree.
 // NOTE: This function clears leftNode/rigthNode recursively and
 // calls _hash() on the given node.
+// TODO: Come back and figure out how to better parallelize this code.
 func (tree *MutableTree) saveNewNodes(version int64) error {
 	nonce := uint32(0)
-	newNodes := make([]*Node, 0)
-	var recursiveAssignKey func(*Node) ([]byte, error)
-	recursiveAssignKey = func(node *Node) ([]byte, error) {
+
+	nodeChan := make(chan *Node, 64) // Buffered channel to avoid blocking.
+	doneChan := make(chan error, 1)  // Channel to signal completion and return any errors.
```

Comment:
There is some overhead with creating channels and goroutine, although likely not so big, it might be better to couple channel and goroutine creation with the lifecycle of `MutableTree` instead of `saveNewNodes`.

Since I don't know off hand how much overhead that actually is maybe it's fine as-is too.